Assets/Scripts/NPCs/Cultist.cs

```diff
@@ -1,205 +1,113 @@
-            Vector2 direction = (xPos < transform.position.x) ? Vector2.left : Vector2.right;
 
-            // Rotate the cultist based on direction
-            if (direction == Vector2.left)
-            {
-                transform.rotation = Quaternion.Euler(0, 180, 0); // Face left
-            }
-            else
-            {
-                transform.rotation = Quaternion.Euler(0, 0, 0); // Face right
-            }
-
-            transform.Translate(direction * _speed * Time.deltaTime, Space.World);
-
-            yield return null;
-        }
+        m_CurrentState = ECultistState.Idle;
+        ChangeState(ECultistState.Idle);
     }
-
-    public void CollectDeadBody(GameObject _deadbody)
+    private void Update()
     {
-        StartCoroutine(CollectDeadBodyCoroutine(_deadbody));
-    }
-    public void Idle()
-    {
-
-        //This requires a bit of fixing
-        // StartCoroutine(IdleRoutine());
+        m_CultistStates[(int)m_CurrentState].UpdateState();
     }
 
-    private IEnumerator CollectDeadBodyCoroutine(GameObject _deadbody)
+    public void ChangeState(ECultistState newState, DeadBody deadbody = null, Transform enemyTransform = null)
     {
-        Transform closestBody = _deadbody.transform;
+        m_CultistStates[(int)m_CurrentState]?.ExitState();
+        m_CurrentState = newState;
 
-        if (closestBody != null)
+        if (newState == ECultistState.Collect && deadbody != null)
         {
-            isBusy = true;
-            state = State.Collect;
-
-            m_Animator.SetTrigger("Sprint");
-
-            yield return MoveToXPositionRoutine(closestBody.position.x);
-            m_Animator.SetTrigger("Idle");
-
-            //If other cultist managed to pick up the body before this one(shouldn't happen, but just in case)
-            if (closestBody == null)
-            {
-                state = State.Idle;
-                isBusy = false;
-                yield break;
-            }
-            // Destroy the dead body once close enough
-            m_Animator.SetTrigger("Collect");
-            yield return new WaitForSeconds(1f);
-
-            closestBody.GetComponent<DeadBody>().FreeToPool();
-            IsCarryingBody = true;
-            state = State.Carry;
-
-            // Enable child component (dead body) of cultist
-            transform.GetChild(0).gameObject.SetActive(true);
-
-            m_Animator.SetTrigger("Carry");
-
-            // Walk towards the grave
-            yield return MoveToXPositionRoutine(_graveyardLocation.position.x); //for now its -10, will put a graveyard position later
-
-            transform.GetChild(0).gameObject.SetActive(false);//Disable the deadbody object
-
-            IsCarryingBody = false;
-
-            // Switch to idle state
-            isBusy = false;
-            state = State.Idle;
-
+            var collectState = (CollectState)m_CultistStates[(int)ECultistState.Collect];
+            collectState.SetDeadBody(deadbody);
```

Comment:
why store it in var and not just do
`((CollectState)m_CultistStates[(int)ECultistState.Collect]).SetDeadBody(deadbody)`
same question applies to the other ifs below